pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -368,7 +368,20 @@ func (t *clusterCache) Key() string {
 		sort.Strings(nv)
 		params = append(params, nv...)
 	}
-	return "cds://" + strings.Join(params, "~")
+	if t.service != nil {
+		params = append(params, string(t.service.Hostname)+"/"+t.service.Attributes.Namespace)
+	}
+	if t.destinationRule != nil {
+		params = append(params, t.destinationRule.Name+"/"+t.destinationRule.Namespace)
+	}
+	params = append(params, t.envoyFilterKeys...)
+
+	hash := md5.New()
+	for _, param := range params {
+		hash.Write([]byte(param))
+	}
+	sum := hash.Sum(nil)
+	return hex.EncodeToString(sum)
```

Comment:
This makes it a bit harder to debug.. maybe ok for performance though.

Reply:
Yes, that's also what I think. I could not balance them.

Reply:
Can we add a debugKey() method that is used in debug logs?

Reply:
We can, but currently after the key is generated, we pass it to xds cache, so we have no chance to call `debugKey`